tsdb/engine/tsm1/engine.go

```diff
@@ -2588,6 +2649,18 @@ func (e *Engine) createVarRefIterator(ctx context.Context, measurement string, o
 		return nil, nil
 	}
 
+	// check for optimized series iteration for tsi index
+	if e.index.Type() == tsdb.TSI1IndexName {
```

Comment:
This has a unit test but I'm not sure there's actually a way to hit it from an influxql query without `Ordered=true`, so might need to remove this section.